clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -376,25 +376,21 @@ protected Map<Node, FetchSessionHandler.FetchRequestData> prepareCloseFetchSessi
         final Cluster cluster = metadata.fetch();
```

Comment:
On a slightly different topic, should `prepareCloseFetchSessionRequests` be synchronized too? We use to have a synchronize in the `close` method but it was removed during the refactoring.

Reply:
I made the `closeInternal()` method `synchronized` as that more closely emulates the synchronization in <= 3.5. It covers `prepareCloseFetchSessionRequests()` too.